guardian/cdk
# chore: Add workflow to combine PRs

## What does this change?
Use https://github.com/github/combine-prs to combine Dependabot PRs to make them more manageable by creating a mega PR of all the open PRs that have passed CI.

The resulting PR will have a `fix` prefix, meaning it'll create a patch release on merge. This is so that any vulnerability fixes in the bumped dependencies (e.g https://github.com/guardian/cdk/pull/1926) go out early, as opposed to waiting for a feature change PR.

## How to test
This new workflow has a `workflow_dispatch` trigger, therefore to test we can run the workflow and observe the resulting PR. Alas, this can only be done once the workflow is on `main`. An example can be found here https://github.com/guardian/master-to-main/pull/268.

## How can we measure success?
Merging Dependabot PRs becomes less of a chore.

## Have we considered potential risks?
N/A. The checks performed in CI will catch any possible regressions. That is, CI helps provide confidence in the resulting combine PR.

## Checklist


[^1]: Consider whether this is something that will mean changes to projects that have already been migrated, or to the CDK CLI tool. If changes are required, consider adding a checklist here and/or linking to related PRs.
[^2]: If you are adding a new construct or pattern, has new documentation been added? If you are amending defaults or changing behaviour, are the existing docs still valid?
